tests/components/smhi/test_config_flow.py

```diff
@@ -200,6 +206,8 @@ async def test_flow_entry_created_from_user_input() -> None:
         return_value={"test": "something", "name_exist": "config"},
     ), patch.object(
         flow, "_check_location", return_value=True
+    ), patch.object(
```

Comment:
Sorry I didn't see it the first pass or I would have pointed it out earlier. I realize this is existing, since it's being touched, we should fix it.  

We don't want tests that patch home assistant internals.  

This test needs to be rewritten to avoid patching flow.

Reply:
See the comments I made earlier to Martin. I intend to rewrite the config flow and thereby the tests too but in the next PR after this one. Is that ok from your perspective or do I need to rewrite a single test now for this to be ok to merge?

Reply:
I don't like merging changes that make the problem worse, but since you are going to rewrite it, I think it is ok in this case.